src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -85,6 +85,16 @@ public static String getFirstNWords(String string, int numOfWords) {
             firstNWords.append(stringArray[i]);
             firstNWords.append(" ");
         }
-        return firstNWords.toString().trim().concat("…");
+        return firstNWords.toString();
+    }
+
+    /**
+     * Gets the first given number of words of the given string with ellipsis appended at the end.
+     * @param string string
+     * @param numOfWords number of words
+     * @return first given number of words of the given string with ellipsis appended at the end
+     */
+    public static String getFirstNWordsWithEllipsis(String string, int numOfWords) {
+        return getFirstNWords(string, numOfWords).trim().concat("…");
```

Comment:
We don't want to add the ... if there are less than N words already right?

Reply:
Oh yes, that is a logic error - fixed in 0576fbd